api/src/main/java/org/apache/iceberg/io/SupportsRecoveryOperations.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.io;
+
+/**
+ * This interface is intended as an extension for FileIO implementations to provide additional
+ * best-effort recovery operations that can be useful for repairing corrupted tables where there are
+ * reachable files missing from disk. (e.g. a live manifest points to data file entry which no
+ * longer exists on disk)
+ */
+public interface SupportsRecoveryOperations {
+
+  /**
+   * Perform a best-effort recovery of a file at a given path
+   *
+   * @param path Absolute path of file to attempt recovery for
+   * @return true if recovery was successful, false otherwise
+   */
+  boolean recoverFile(String path);
```

Comment:
Any thoughts on having the signature be 
**recoverFile(String path, Throwable exeception)**

Reply:
Is there a specific use case you had in mind here?  In practice, I've always found that either you have the versioned object to recover or you don't.  There may be some cases where there's a permission issue or a reason that the operation can't be invoked, but that tends to be rare.

Reply:
Nothing in mind, just feels like it would be better to add it now than later. I was thinking the easiest thing would be being able to fail fast if the error is something the IO knows it can't recover from. If no one uses it then it shouldn't be a huge expense either.

Reply:
I don't think the intent is to use this in a normal read path, more like a procedure (the rational I cover in an earlier comment).  Since this is a "best effort" operation, I feel it would be adding unnecessary complexity that we can address later if there's a need (e.g. we can always extend with default implementations in the future).

Reply:
I'm just thinking in generalities since I am not familiar with the planned implementations so I'm ok with that.

Reply:
Sure, I think this interface generalizes in the sense that it provides a common method for recovering a file that different FileIOs can implement. 

 Now does it generalize beyond the Repair action itself? I can see the interface being used as a primitive if someone doesn't want to do the full repair action. For instance if a user already knows a single corrupted manifest with entries of data files some of which may/may not be on disk, then they can just write their own logic to recover it via this interface.